elyra/pipeline/component.py

```diff
@@ -44,12 +44,12 @@ def __init__(self, ref: str, name: str, type: str, value: str, description: str,
         :param required: Whether the property is required
         """
 
-        if not ref:
-            raise ValueError("Invalid component: Missing field 'ref'.")
+        if not id:
+            raise ValueError("Invalid component: Missing field 'id'.")
         if not name:
             raise ValueError("Invalid component: Missing field 'name'.")
 
-        self._ref = ref
+        self._ref = id
```

Comment:
Shouldn't the attribute and property names be updated to reflect `id` rather than `ref`?